stdlib/universe/map.go

```diff
@@ -174,23 +161,15 @@ func (t *mapTransformation) Process(id execute.DatasetID, tbl flux.Table) error
 				// Determine on which cols to group
 				on = make(map[string]bool, len(tbl.Key().Cols()))
 				for _, c := range tbl.Key().Cols() {
-					on[c.Label] = t.mergeKey || execute.ContainsStr(keys, c.Label)
+					on[c.Label] = execute.ContainsStr(keys, c.Label)
```

Comment:
Why are we still checking for the columns to group given that `map` cannot remove columns anymore?  
The group key should remain the same.  

I think that `on` is useless now ad you can directly use the group key of the original table in the util function `groupKeyForObject`. 
You still need to check the values because it can cause regrouping.

Reply:
It is still possible to remove columns if the function does not use `with`. 

I have added  a test for that case.

Reply:
Definitely right.